app/models/service_template_ansible_playbook.rb

```diff
@@ -50,19 +50,46 @@ def self.prepare_job_template_and_dialog(action, service_name, description, conf
   end
   private_class_method :prepare_job_template_and_dialog
 
-  def self.create_job_template(name, description, info)
-    playbook = ManageIQ::Providers::AnsibleTower::ConfigurationManager::Playbook.find(info[:playbook_id])
-    # tower = playbook.manager
+  def self.create_job_templates(service_name, description, config_info, auth_user)
+    {}.tap do |hash|
+      [:provision, :retirement, :reconfigure].each do |action|
+        next unless config_info[action]
+        hash[action.to_sym] = { :configuration_template => create_job_template("#{service_name}_#{action}", description, config_info[action], auth_user) }
```

Comment:
Creating job template is the first step. We don't have any id at this stage. Having the service name as part of job template name is better for debugging. If we don't want random number we could try sequential number until the tower no longer rejects. Random number is probably better for performance's sake.